metadetect/lsst/measure.py

```diff
@@ -111,6 +111,7 @@ def detect_and_deblend(
     # the variance
     # detection_config.thresholdType = 'variance'
     detection_config.thresholdValue = thresh
+    detection_config.thresholdType = 'stdev'
```

Comment:
This change seems non-trivial since we go from a fixed threshold of 5 to something that looks dynamic. @arunkannawadi can you explain what the difference is? 

@esheldon you should take a look here.

Reply:
`stdev` means that you apply a 5sigma threshold, where sigma is the standard deviation of all the good pixels. `pixel_stdev` would mean you apply a 5sigma threshold, where sigma would be the per-pixel value from the variance plane (its square root). When you ran the tests about a year ago, the `thresholdType` by default was `stdev` but it changed to `pixel_stdev` more recently. So this is not actually a change, but undoes a change made in the LSST stack. Unless you had explicitly set `thresholdType` to `value` somewhere, it would have not been a fixed threshold.

Reply:
which stack release has that change?

Reply:
w_2023_39

Reply:
My notes say using `variance` meant using the square root of the variance

Reply:
Anyway, by not specifying the `thresholdType`, you had been using the default value, which was `stdev` for the past 12 years (!!!) until `w_2023_39`.

Reply:
As for 1. that is bad.  We do not want the objects included in the standard deviation

Reply:
I know it is "clipped" but that's still going to be biased.

Reply:
It's a clipped standard deviation, if that helps.

Reply:
I just want to note that using the true variance plane did not work.  This was thought to be a bug, see https://jira.lsstcorp.org/browse/DM-32390

this would be our preferred method, since we know it from the standard deviation of the noise image

Reply:
You probably want `pixel_stdev` (the current default) and fix th test. The current test setup does not trigger the AllZeroWeightsError

Reply:
All other tests pass. But given that it passed lol the extensive tests with sims, how biased could it be? What effect that are not in the sims could make the bias worrisome?

Reply:
Wait, are you talking about the ngmix-upstream test? That's on the `main` branch of `ngmix`.

Reply:
Right. If there is an interaction between this change, the DM code, and ngmix on main, we should understand that.

Reply:
This change has nothing to do with those two failures.

- `master` on `metadetect` + latest weekly on `stackvana` (`w_2024_02` on linux) + latest `ngmix` on PyPI (`v2.3.0`) - 1 failure which I have narrowed down to the change in LSST Science Pipelines.
- `master` on `metadetect` + latest weekly on `stackvana` (`w_2024_02` on linx) + `master` on `ngmix` - 2 additional failures.

Those two additional failures have to be due to changes in `ngmix` since the last release. I don't know how the CI works there, but it's not suprising given that pushes to `ngmix` don't trigger the `lsst` tests here.